app/src/main/java/unipd/se18/ocrcamera/cameraActiviti.java

```diff
@@ -0,0 +1,701 @@
+                    editor.commit();
+
+                    //@author Leonardo Rossi
+                    //Temporary stores the caprured photo into a file that will be used from the Camera Result activity
+                    Bitmap bmp = BitmapFactory.decodeByteArray(photoByteArray, 0, photoByteArray.length);
+                    String filePath= tempFileImage(CameraActivity.this, bmp,"capturedImage");
+
+                    //An intent that will launch the activity that will analyse the photo
+                    Intent i = new Intent(CameraActivity.this, ResultActivity.class);
+                    i.putExtra("imageDataPath", filePath);
+                    startActivity(i);
+                }
+            };
+            mImageReader.setOnImageAvailableListener(readerListener, mBackgroundHandler);
+
+            //@author Leonardo Rosi
+
+            //Output surfaces
+            List<Surface> outputSurface = new ArrayList<>(2);
+            outputSurface.add(mImageReader.getSurface());
+            outputSurface.add(new Surface(mCameraTextureView.getSurfaceTexture()));
+
+            //Creation of a capture request to take a photo from the camera
+            final CaptureRequest.Builder captureBuilder = mCameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_STILL_CAPTURE);
+            captureBuilder.addTarget(mImageReader.getSurface());
+            captureBuilder.set(CaptureRequest.CONTROL_MODE, CameraMetadata.CONTROL_MODE_AUTO);
+
+            //Check orientation base on device
+            int rotation = getWindowManager().getDefaultDisplay().getRotation();
+            captureBuilder.set(CaptureRequest.JPEG_ORIENTATION,ORIENTATIONS.get(rotation));
+
+            final CameraCaptureSession.CaptureCallback captureListener = new CameraCaptureSession.CaptureCallback() {
+                @Override
+                public void onCaptureCompleted(@NonNull CameraCaptureSession session,
+                                               @NonNull CaptureRequest request, @NonNull TotalCaptureResult result) {
+                    super.onCaptureCompleted(session, request, result);
+                    createCameraPreview();
+                }
+            };
+
+            mCameraDevice.createCaptureSession(outputSurface, new CameraCaptureSession.StateCallback() {
+                @Override
+                public void onConfigured(@NonNull CameraCaptureSession cameraCaptureSession) {
+                    try {
+                        cameraCaptureSession.capture(captureBuilder.build(),captureListener, mBackgroundHandler);
+                    } catch (CameraAccessException e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                @Override
+                public void onConfigureFailed(@NonNull CameraCaptureSession cameraCaptureSession)
+                {
+                }
+            },mBackgroundHandler);
+        } catch(CameraAccessException e) {
```

Comment:
Rather than enclosing all the code inside the try-catch block, you should insert inside the try-catch only the code that can throw an exception, so you can properly manage it.
You can check the doc:
https://developer.android.com/reference/android/hardware/camera2/CameraAccessException
too see which methods can throw a CameraAccessException and under what conditions.